ctx.py

```
@@ -12,21 +12,102 @@
-    if not args or not hasattr(args, 'cfg_dir') or not getattr(args, 'cfg_dir'):
-        return None
+class ConfigKey(enum.Enum):
+    CONTEXT = 'ctx'
+    TERMINAL = 'terminal'
+
+    @classmethod
+    def has_key(cls, key: str):
+        return key in [e.value for e in cls]
+
+
+class ContextConfig(object):
+    def __init__(self):
+        self.raw_cfg = {}
+
+    def _value_or_none(self, key: ConfigKey, attribute_name: str):
+        if self.raw_cfg.get(key.value):
+            return self.raw_cfg.get(key.value).get(attribute_name)
+        else:
+            return None
+
+    def _set_value(self, key: ConfigKey, attribute_name: str, value):
+        if self.raw_cfg.get(key.value) is None:
+            self.raw_cfg[key.value] = {attribute_name: value}
+        else:
+            self.raw_cfg.get(key.value)[attribute_name] = value
+
+    def context_value_or_none(self, attribute_name: str):
+        print(self.raw_cfg)
+        return self._value_or_none(ConfigKey.CONTEXT, attribute_name)
+
+    def set_context_value(self, attribute_name: str, value):
+        self._set_value(ConfigKey.CONTEXT, value)
+
+    def terminal_value_or_none(self, attribute_name: str):
+        return self._value_or_none(ConfigKey.TERMINAL, attribute_name)
+
+    def set_terminal_value(self, attribute_name: str, value):
+        self._set_value(ConfigKey.TERMINAL, attribute_name, value)
+
+
+def load_config_from_file():
+    # Load $HOME/.cc-utils.cfg if it exists
+    config_file = Path.home() / '.cc-utils.cfg'
+    if config_file.exists() and config_file.is_file():
+        try:
+            cfg = util.parse_yaml_file(config_file)
 
+            for cfg_key in cfg:
+                if not ConfigKey.has_key(cfg_key):
+                    continue
+
+                for k, v in cfg.get(cfg_key).items():
+                    config._set_value(ConfigKey(cfg_key), k, v)
+
+        except Exception as e:
+            util.info(str(e))
```

warn - OR: let it propagate